docs/lab/07/index.md

```diff
@@ -0,0 +1,263 @@
+---
+sidebar_position: 7
+description: Inter-Integrated Circuit
+slug: /lab/07
+---
+
+# 07 - Inter-Integrated Circuit
+
+This lab will teach you how to communicate with hardware devices using the Inter-Integrated Circuit (I2C) protocol, in Embassy.
+
+## Resources
+
+TBD
+
+## Inter-Integrated Circuit (I2C)
+
+The Inter-Integrated Circuit (I2C) is a synchronous, multi-controller/multi-target communication protocol. Similarly to the SPI, it allows data transfer between a controller and one or more peripheral ICs, but it uses only 2 wires (1 data line and 1 clock line, making it half-duplex) and has a different way of addressing the peripherals: using the their unique *addresses*.
+
+### Configuration
+
+![I2C_protocol](images/i2c_network.svg)
+
+I2C transmission uses 2 lines:
+
+- **SCL** - **S**erial **CL**ock line - clock is generated by the controller - used to synchronize communication between the controller and the targets
+- **SDA** - **S**erial **DA**ta line - carries data between the controller and the addressed target
+    - targets read data from SDA **only** when the clock is *low*
+    - targets write data to SDA **only** when the clock is *high*
+
+:::info
+The communication is *half-duplex*. This means that data is transmitted only in one direction at a time, since there is only *one* data line that can be used both for sending data to the target and receiving data from the target.
+:::
+
+:::info
+The `SDA` and `SCL` wires are never actually driven (set to `LOW`/`HIGH`) by the controller/peripherals. The line is controlled by either pulling the line low or releasing the line high. 
+
+When the line is *pulled down*, this means that it is tied to `GND` through a pull-down resistor. This electronically translates to `LOW`.
+
+When the line is *released*, or *pulled up*, this means that it ties back to `3V3` (which we can consider as being the "default" state of the wire) through a pull-up resistor. This electronically translates to `HIGH`.
+
+The pull-down resistor has a smaller resistance than the pull-up one, so every time a peripheral wants to write a 0 to the line, it changes the circuit of the bus so that it ties directly to ground.
+
+This is called *open-drain connection*. You can read more about how it works [here](https://www.ti.com/lit/an/sbaa565/sbaa565.pdf?ts=1712891793335#:~:text=I2C%20is%20a%20two%2Dwire,and%20receive%20commands%20and%20data.), at section 2.2.
```

Comment:
Open drain uses only pull up resistor and connects directly to the ground for logic LOW. Pull down resistors are used for push-pull configuration(ex: spi) and is recommanded only if there is another CS pin